scipy/optimize/_differentialevolution.py

```diff
@@ -757,11 +833,21 @@ def solve(self):
                 self.population_energies[0] = result.fun
                 self.population[0] = self._unscale_parameters(result.x)
 
+        if self.constraints:
+            DE_result.constr = [c.violation(DE_result.x) for
+                                c in self.constraints]
+            DE_result.constr_violation = np.max(
+                np.concatenate(DE_result.constr))
+            DE_result.maxcv = DE_result.constr_violation
+            if DE_result.maxcv > 0:
+                # if the result is infeasible then success must be False
+                DE_result.success = False
```

Comment:
`DE_result.success = False` never happens in tests

Reply:
A test could be added where it's going to be impossible for constraints to be satisfied.

Reply:
Added a test for this.